### PR TITLE
EVEREST-832 Fix access to namespaces

### PR DESCRIPTION
**Pull request by recharte**
[EVEREST-832](https://jira.percona.com/browse/EVEREST-832) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**Fix access to namespaces**
---
**Problem:**
EVEREST-832

DBCs and DBBs had access to all BackupStorages and MonitoringConfigs regardless of the namespace.

**Cause:**
We weren't validating if a given BackupStorage or MonitoringConfig was allowed in the DBC's namespace.

**Solution:**
Validate if a given BackupStorage or MonitoringConfig is allowed in the DBC's namespace.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?

**Tests**
- [ ] Is an Integration test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

[EVEREST-832]: https://perconadev.atlassian.net/browse/EVEREST-832?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/validation.go [hunk 1]
```diff
@@ -606,15 +604,50 @@ func validateBackupStoragesFor( //nolint:cyclop
...
+		if ns == namespace {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("monitoring config %s is not allowed for namespace %s", name, namespace)
+	}
+
+	return mc, nil
```
Comment (oksana-grishchenko) [inline api/validation.go:650]:
```suggestion
	for _, ns := range mc.Spec.TargetNamespaces {
		if ns == namespace {
			return mc, nil
		}
	}
	return nil, fmt.Errorf("monitoring config %s is not allowed for namespace %s", name, namespace)
```

File: api/validation.go [hunk 2]
```diff
@@ -606,15 +604,50 @@ func validateBackupStoragesFor( //nolint:cyclop
...
+		if ns == namespace {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("backup storage %s is not allowed for namespace %s", name, namespace)
+	}
+
+	return bs, nil
```
Comment (oksana-grishchenko) [inline api/validation.go:627]:
```suggestion
	for _, ns := range bs.Spec.TargetNamespaces {
		if ns == namespace {
			return bs, nil
		}
	}
	return nil, fmt.Errorf("backup storage %s is not allowed for namespace %s", name, namespace)
```

File: api/validation.go [hunk 3]
```diff
@@ -552,16 +549,17 @@ func (e *EverestServer) validateDatabaseClusterCR(ctx echo.Context, namespace st
 
 func validateBackupStoragesFor( //nolint:cyclop
 	ctx context.Context,
+	namespace string,
 	databaseCluster *DatabaseCluster,
-	validateBackupStorageAccessFunc func(context.Context, string) (*everestv1alpha1.BackupStorage, error),
+	validateBackupStorageAccessFunc func(context.Context, string, string) (*everestv1alpha1.BackupStorage, error),
```
Comment (oksana-grishchenko) [inline api/validation.go:554]: please update the unit-tests for this change

**Review by oksana-grishchenko — APPROVED** [review]

File: api/validation.go (hunk 3 above)
Comment (recharte) [reply api/validation.go:554]: Done :rocket:

**Pull request merged**